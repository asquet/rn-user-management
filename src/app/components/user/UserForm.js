import React from 'react';
import { View, Picker } from 'react-native';
import Button from '../common/content/Button';
import DatePicker from '../common/form/DatePicker';
import ImagePicker from '../common/form/ImagePicker';
import Autocomplete from '../common/form/Autocomplete';
import InputRow from '../common/form/InputRow';
import { formStyles } from '../../stylesheet';

export default function UserForm({ isNew, user, onChange, onSave, onCancel, managers, positions }) {
  return (
    <View style={formStyles.container}>
      <InputRow
        label="Name"
        value={user.name}
        onChange={val => onChange('name', val)}
      />

      <InputRow
        label="Birth date"
      >
        <DatePicker
          date={user.birthDate}
          mode="date"
          placeholder="Select Birth date"
          format="YYYY-MM-DD"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          onDateChange={date => onChange('birthDate', date)}
        />
      </InputRow>
      <InputRow
        label="User Pic"

        onChange={val => onChange('userPic', val)}
      >
        <ImagePicker value={user.userPic} />
      </InputRow>
      <InputRow
        label="Manager"
      >
        <Autocomplete
          placeholder="Manager not selected"
          options={managers.map(m => ({ value: m.id, label: m.name }))}
          onChangeValue={val => onChange('managerId', val)}
          selection={user.managerId}
        />
      </InputRow>
      <InputRow
        label="Position"
        onChange={val => onChange('positionId', val)}
      >
        <Picker
          selectedValue={user.positionId}
          onValueChange={positionId => onChange('positionId', positionId)}
        >
          {
            positions.map(position => (
              <Picker.Item label={position.name} value={position.id} key={position.id} />
            ))
          }
        </Picker>
      </InputRow>

      <View style={formStyles.controlsWrapper}>
        <Button
          onPress={() => onSave(user)}
          title="Save"
          buttonStyle={formStyles.controlsSave}
        />
        <Button
          onPress={onCancel}
          title="Cancel"
          buttonStyle={formStyles.controlsCancel}
        />
      </View>
    </View>
  );
}

UserForm.propTypes = {
  isNew: React.PropTypes.bool.isRequired,
  user: React.PropTypes.shape({
    name: React.PropTypes.string,
    birthDate: React.PropTypes.string,
    userPic: React.PropTypes.string,
    managerId: React.PropTypes.number,
    positionId: React.PropTypes.number,
  }).isRequired,
  onChange: React.PropTypes.func.isRequired,
  onSave: React.PropTypes.func.isRequired,
  onCancel: React.PropTypes.func.isRequired,
  managers: React.PropTypes.arrayOf(React.PropTypes.shape({})).isRequired,
  positions: React.PropTypes.arrayOf(React.PropTypes.shape({})).isRequired,
};
