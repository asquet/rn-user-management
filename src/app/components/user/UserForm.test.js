import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserForm from './UserForm';

// Note: test renderer must be required after react-native.

describe('UserForm', () => {
  const positions = [{ id: 1, name: '1' }];
  const managers = [{ id: 1, name: '1' }];
  const user = { name: 'user', birthDate: '2000-10-10' };
  it('renders correctly', () => {
    const tree = renderer.create(
      <UserForm
        positions={positions}
        onSave={jest.fn()}
        onChange={jest.fn()}
        isNew managers={managers}
        onCancel={jest.fn()}
        user={user}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
