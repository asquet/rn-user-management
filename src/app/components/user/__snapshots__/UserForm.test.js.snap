// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Create user
  </Text>
  <View
    style={
      Array [
        Object {
          "height": 50,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Name"
      style={
        Object {
          "flex": 1,
        }
      }
      value="user"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "height": 50,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "width": 142,
            },
            Object {
              "flex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#aaa",
                "borderWidth": 1,
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
              },
              undefined,
              false,
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#333",
                },
                undefined,
              ]
            }
          >
            2000-10-10
          </Text>
        </View>
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 32,
                "marginLeft": 5,
                "marginRight": 5,
                "width": 32,
              },
              undefined,
            ]
          }
        />
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          supportedOrientations={
            Array [
              "portrait",
              "portrait-upside-down",
              "landscape",
              "landscape-left",
              "landscape-right",
            ]
          }
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "alignItems": "flex-end",
                    "backgroundColor": "#00000077",
                    "flex": 1,
                    "flexDirection": "row",
                  },
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "height": 0,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderTopColor": "#ccc",
                          "borderTopWidth": 1,
                          "marginTop": 42,
                        },
                        undefined,
                      ]
                    }
                  >
                    <RCTDatePicker
                      date={971118000000}
                      maximumDate={undefined}
                      minimumDate={undefined}
                      minuteInterval={undefined}
                      mode="date"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 216,
                        }
                      }
                      timeZoneOffsetInMinutes={undefined}
                    />
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 42,
                            "justifyContent": "center",
                            "padding": 20,
                            "position": "absolute",
                            "top": 0,
                          },
                          Object {
                            "left": 0,
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#46cf98",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#666",
                          },
                          undefined,
                        ]
                      }
                    >
                      Cancel
                    </Text>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 42,
                            "justifyContent": "center",
                            "padding": 20,
                            "position": "absolute",
                            "top": 0,
                          },
                          Object {
                            "right": 0,
                          },
                          undefined,
                        ],
                      ]
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#46cf98",
                            "fontSize": 16,
                          },
                          undefined,
                        ]
                      }
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 50,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Select image
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 50,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "zIndex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#b9b9b9",
                "borderRadius": 1,
                "borderWidth": 1,
              },
              undefined,
            ]
          }
        >
          <TextInput
            blurOnSubmit={true}
            data={Array []}
            defaultValue=""
            onBlur={[Function]}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            onFocus={[Function]}
            placeholder="Manager not selected"
            renderItem={[Function]}
            renderSeparator={[Function]}
            renderTextInput={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "height": 40,
                  "paddingLeft": 3,
                },
                undefined,
              ]
            }
            value=""
          />
        </View>
        <View
          style={undefined}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 50,
          "padding": 3,
        },
        null,
      ]
    }
  >
    <View
      style={undefined}
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "1",
              "textColor": undefined,
              "value": 1,
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#55dd66",
          "borderRadius": 3,
          "borderWidth": 1,
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "steelblue",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Save
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#dd5555",
          "borderRadius": 3,
          "borderWidth": 1,
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "steelblue",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;
